x/evm/types/types.go

```diff
@@ -1041,3 +1028,60 @@ func createTransferMultisigParams(addrs []common.Address, threshold uint8) ([]by
 
 	return result, nil
 }
+
+// ValidateBasic does stateless validation of the object
+func (m *BurnerInfo) ValidateBasic() error {
+	if m.DestinationChain == "" {
+		return fmt.Errorf("destination chain not set")
+	}
+	if m.Asset == "" {
+		return fmt.Errorf("asset not set")
+	}
+	if m.Symbol == "" {
+		return fmt.Errorf("symbol not set")
+	}
+
+	return nil
+}
+
+// ValidateBasic does stateless validation of the object
+func (m *Gateway) ValidateBasic() error {
+	if m.Status == GatewayStatusNone {
+		return fmt.Errorf("gatway status not set")
+	}
+	return nil
+}
+
+// ValidateBasic does stateless validation of the object
+func (m *ERC20TokenMetadata) ValidateBasic() error {
+	if m.Status == NonExistent {
+		return fmt.Errorf("token status not set")
+	}
+
+	if m.Asset == "" {
+		return fmt.Errorf("asset not set")
+	}
+
+	if m.ChainID.IsNil() || !m.ChainID.IsPositive() {
+		return fmt.Errorf("chain ID not set")
+	}
+
+	if err := m.Details.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
this will need to be updated to also validate the minimum amount to mint/withdrawal

Reply:
ok, let me know if that PR is merged in before this one

Reply:
it was already merged.